jdaviz/core/template_mixin.py

```diff
@@ -1888,6 +1888,11 @@ def add_results_from_plugin(self, data_item, replace=None, label=None):
         Add ``data_item`` to the app's data_collection according to the default or user-provided
         label and adds to any requested viewers.
         """
+
+        # Note that we can only preserve one of percentile or vmin+vmax
+        preserve_attributes = ("color", "alpha", "bias", "linewidth", "stretch",
+                               "v_min", "v_max", "cmap")
+
```

Comment:
Is there any way to access these programmatically since different layer-types might have different options (thinking for image layers created in cubeviz, or scatter layers in lcviz where keeping this list up-to-date with upstream changes might be difficult and if we are ever short any entries, that could be quite confusing to the user).  Maybe using `layer.state.as_dict()` and just hardcoding certain attributes to ignore instead (like the link to `layer.state.layer`)?

Reply:
There are enough attributes in `state` that I think listing the ones to ignore would be more annoying than listing the ones we want. I do see the annoyance with needing to potentially update this (and I didn't think of scatter viewers in this list), but I don't see a good programmatic way to do it that is less annoying.

Reply:
`as_dict` contains a lot less entries than `dir(layer.state)` though... we might only need to exclude `['attribute', 'layer']`

Reply:
It's still a pretty big list, and there's some stuff in there that I'm either not sure what it does and/or wasn't sure if we want to preserve it, e.g. `c_min`, `c_max`, `levels` - are these all contour things? And if so, do we want to preserve contour levels? I rate that as a solid maybe. Maybe it is alright to default to keeping everything but the two you mentioned though.

Reply:
Preserving contours might be a PO question.

Reply:
I think you can safely preserve everything (including contour levels).
I do not know why 'attribute' and 'layer' should not be preserved, but I trust you all on this.